api/v1alpha1/types.go

```diff
@@ -52,12 +52,14 @@ type Schema struct {
 	// and create the CRD for the resourcegraphdefinition.
 	//
 	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Pattern=`^[A-Z][a-zA-Z0-9]*$`
```

Comment:
I believe CRD kind are supposed to match https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apimachinery/pkg/util/validation/validation.go#L244 with an addition length limit of 63 characters.

https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/validation/validation.go#L762

Reply:
TIL, thanks Walter!